connect_ext_test.go

```diff
@@ -183,14 +183,28 @@ func TestServer(t *testing.T) {
 				connect.CodeOf(stream.Err()),
 				connect.CodeInvalidArgument,
 			)
+			assert.Nil(t, stream.Close())
 		})
 		t.Run("count_up_timeout", func(t *testing.T) {
 			ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(-time.Second))
-			defer cancel()
+			t.Cleanup(cancel)
 			_, err := client.CountUp(ctx, connect.NewRequest(&pingv1.CountUpRequest{Number: 1}))
 			assert.NotNil(t, err)
 			assert.Equal(t, connect.CodeOf(err), connect.CodeDeadlineExceeded)
 		})
+		t.Run("count_up_cancel_after_first_response", func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			request := connect.NewRequest(&pingv1.CountUpRequest{Number: 5})
+			request.Header().Set(clientHeader, headerValue)
+			stream, err := client.CountUp(ctx, request)
+			assert.Nil(t, err)
+			assert.True(t, stream.Receive())
+			cancel()
+			assert.False(t, stream.Receive())
```

Comment:
This seems racy and could possibly cause timing-sensitive test flakes.

The issue is that `stream.Receive` will try to synchronously read from the response body. However the `cancel` call above is totally async (the request will notice the cancellation and try to then cancel the HTTP operation, but it is effectively happening in the background, asynchronously.

Perhaps better is to call `Receive` in a loop until it fails with a cancelled error, instead of assuming the very next call will fail. If you get EOF after all 5 messages and no cancelled error, then the test case can fail. _Or_ you could insert reasonable delay to make sure background cancellation has had a chance to execute, like `time.Sleep(100*time.Millisecond)`.

Reply:
The client explicitly looks at context cancellation before calling Receive, behaviour is good to keep that we always do a context check before `Receive()`: https://github.com/connectrpc/connect-go/blob/d9b61e93e4e7e8c2a4c6be7a31fee7c33e7dc9a3/duplex_http_call.go#L166